app/controllers/recipes_controller.rb

```diff
@@ -22,7 +22,7 @@ def create
     @recipe = Recipe.new(recipe_params)
```

Comment:
I think this is how I might implement this.  First it takes the 'categories' param which is a string of comma separated category names and converts that to an array of category name strings.

Then it uses a method on the Category model for finding/creating categories from a list of names, returning an array of Category model instances.  It takes that new array and merges that into the recipe_params hash (essentially replacing the 'categories' param) which is used to initialize the Recipe before saving.

The reason why `merge` is used here is because `recipe_params` is a method call that rebuilds the params hash each time, so any changes to it would be lost.  `merge` is just taking two hashes and smashing them together to a new hash.
```suggestion
    category_names = recipe_params.fetch('categories', '').split(',').map(&:strip)

    @recipe = Recipe.new(recipe_params.merge( 'categories' => Category.from_names(category_names) }))
```

Similar logic could be used for the update route as well.  Although, you'd need to figure out how to populate your form element with the recipe's current categories when it loads so that the user doesn't always have to enter all the categories.

Further refactoring, I'd possibly update the Recipe model to handle category_names itself so that the seeds, controllers and other calling code don't need to be responsible for any of that.

Reply:
This is a comment for both this and the model function of save categories. The reason I had @recipe.save_categories, (I did forget to add save in the model, but that was the intention). Going back into the model, then yea it definitely is not working like I imagine it would. I thought on submitting recipe it would go to the controller call `@recipe.save_categories` and in the model, it would map categories submitted by the form. Not categories already in the table and it would cross reference with the Category table and either find or create the category and save the recipe accordingly.